gradle-baseline-java/src/main/groovy/com/palantir/baseline/plugins/BaselineClassUniquenessPlugin.java

```diff
@@ -27,14 +27,16 @@
  * The task only fails if it finds classes which have the same name but different implementations.
  */
 public class BaselineClassUniquenessPlugin extends AbstractBaselinePlugin {
-
     @Override
     public final void apply(Project project) {
+        BaselineClassUniquenessRule rule = new BaselineClassUniquenessRule(project);
+
+        project.getTasks().addRule(rule);
+
         project.getPlugins().withId("java", plugin -> {
-            project.getTasks().create("checkClassUniqueness", CheckClassUniquenessTask.class, task -> {
-                task.setConfiguration(project.getConfigurations().getByName("runtime"));
-                project.getTasks().getByName("check").dependsOn(task);
-            });
+            rule.apply("checkRuntimeClassUniqueness");
```

Comment:
Needed to do ^ to get the `register` to work (if I change `register` to `create`, I don't need the `rule.apply`)